File: 总题库/1TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) ->List[int]:
        seen = {};
        
        for index, item in enumerate(nums):
            if target - item in seen:
                return [seen[target - item], index]
            else:
                seen[item] = index
                
        return []

File: 总题库/1TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        int numsLen = nums.size();

        for (int i = 0; i < numsLen; i++) {
            if (seen.count(target - nums[i])) {
                return vector<int>({seen[target - nums[i]], i});
            }
            else {
                seen[nums[i]] = i;
            }
        }
        return vector<int>();
    }
};